filebrowser_safe/views.py

```diff
@@ -173,30 +180,47 @@ def browse(request):
                     results_var["select_total"] += 1
             except OSError:
                 # Ignore items that have problems
-                continue
+                return
             else:
-                files.append(fileobject)
+                sort_values[fileobject.filename] = getattr(fileobject, query["o"])
+                files_to_sort.append(fileobject)
                 results_var["results_current"] += 1
 
-        # COUNTER/RESULTS
-        if fileobject.filetype:
-            counter[fileobject.filetype] += 1
+    for i in range(0, file_list_size, batch_size):
+        file_list = file_list[i : i + batch_size]
+        for file in file_list:
+            # EXCLUDE FILES MATCHING ANY OF THE EXCLUDE PATTERNS
+            filtered = not file or file.startswith(".")
+            for re_prefix in filter_re:
+                if re_prefix.search(file):
+                    filtered = True
+            if filtered:
+                continue
+            proc_file = threading.Thread(
+                target=create_file_object,
+                args=(
+                    file,
+                    files_to_sort,
+                ),
+            )
+            proc_file.start()
+            proc_list.append(proc_file)
+        for fo in proc_list:
+            fo.join()
```

Comment:
👍 Batching + threading.  This is really cool, I enjoyed reading up on the `threading` module.  We have a `join` here, so it should be good to go